src/main/java/org/apache/fineract/api/OperationsDetailedApi.java

```diff
@@ -230,6 +218,64 @@ public Page<TransactionRequest> transactionRequests(
         }
     }
 
+    /**
+     * Filter the [TransactionRequests] based on multiple type of ids
+     * @param response instance of HttpServletResponse
+     * @param page the count/number of page which we want to fetch
+     * @param size the size of the single page defaults to [10000]
+     * @param sortedOrder the order of sorting [ASC] or [DESC], defaults to [DESC]
+     * @param filterBy type of filter we want to apply, @see [Filter]
+     * @param ids the list of ids to match the query with
+     * @throws IOException in case of csv writing can cause this exception
+     */
+    @PostMapping("/transactionRequests/export")
+    public void filterTransactionRequests(
+            HttpServletResponse response,
+            @RequestParam(value = "page", required = false, defaultValue = "0") Integer page,
+            @RequestParam(value = "size", required = false, defaultValue = "10000") Integer size,
+            @RequestParam(value = "sortedOrder", required = false, defaultValue = "DESC") String sortedOrder,
+            @RequestParam(value = "filterBy", required = true, defaultValue = "TRANSACTIONID") String filterBy,
+            @RequestBody(required = true) List<String> ids) throws IOException {
+
+        Specifications<TransactionRequest> spec = null;
+        Filter filter;
+        try {
+            filter = parseFilter(filterBy);
+            logger.info("Filter parsed successfully " + filter.name());
+        } catch (Exception e) {
+            filter = Filter.TRANSACTIONID;
+            logger.info("Filter parsing failed. Setting default filter " + filter.name());
+        }
+        switch (filter) {
+            case TRANSACTIONID:
+                spec = TransactionRequestSpecs.in(TransactionRequest_.transactionId, ids);
+                break;
+            case PAYERID:
+                spec = TransactionRequestSpecs.in(TransactionRequest_.payerPartyId, ids);
+                break;
+            case PAYEEID:
+                spec = TransactionRequestSpecs.in(TransactionRequest_.payeePartyId, ids);
+                break;
+            case WORKFLOWINSTANCEKEY:
+                spec = TransactionRequestSpecs.in(TransactionRequest_.workflowInstanceKey, ids);
+                break;
+        }
+        PageRequest pager = new PageRequest(page, size, new Sort(Sort.Direction.valueOf(sortedOrder), "startedAt"));
+
+
+        Page<TransactionRequest> result;
+        if(spec == null) {
+            result = transactionRequestRepository.findAll(pager);
+        } else {
+            result = transactionRequestRepository.findAll(spec, pager);
+        }
+        CsvUtility.writeToCsv(response, result.getContent());
```

Comment:
Please add some error handling logic here to convert 500 internal servererror on IOException to a standard JSON error body with code and description.

Reply:
noted, will update this PR.

Reply:
CsvWriter has 
embedded error handling. It throws `WriteToCsvException` which has all the relevant information 
```
String developerMessage;
String errorDescription;
ErrorCode errorCode;
```

here ErrorCode is below set of custom codes
enum ErrorCode {
        CSV_GET_WRITER,
        CSV_WRITE_HEADER,
        CSV_WRITE_DATA,
        CSV_STREAM,
        CSV_BUILDER
    }